resources/views/components/button.blade.php

```diff
@@ -1,3 +1,2 @@
-<button {{ $attributes->merge(['type' => 'submit']) }}>
-    {{ $slot }}
-</button>
+<button {{ $attributes->merge(['type' => $type]) }}>{{ $slot }}</button>
```

Comment:
I'm not sure why the original code wasn't working. Looking at the Laravel docs for [Non-class attribute merging](https://laravel.com/docs/9.x/blade#non-class-attribute-merging), it looks like it was correct.

Also what's the need for the button blade component, as opposed to just using the `<button>` element directly?

Reply:
The original code wasn't working because I made the mistake of including an attribute in the component class and setting it in the constructor while also trying to use attribute merging. I think you're right in that there's no advantage to having this as a standalone component, though, @jobara. I'll open an issue to deprecate and remove it for v3.

Reply:
Simplified to use standard non-class attribute merging in [a8608df](https://github.com/fluid-project/hearth/pull/122/commits/a8608dfb054598c5084734a235b30a31655da8e2).